### ISSUE TITLE
New endpoint that returns the transactions paginated for an identity

### ISSUE THREAD

**Issue by sallymoc** [qubic/go-archiver#63]

**Comment by sallymoc** [comment]
Query parameters:
- from_date and to_date: Specify the date range for transactions. 
- from_tick and to_tick: Set the block range for the transactions you're interested in. If both from_date and from_tick (or to_date and to_tick) are provided, the block parameter takes precedence.
If 'from_date' and 'from_tick’ are provided, 'from_tick’ will be used.
If 'to_date' and 'to_tick’ are provided, 'to_tick’ will be used.
- cursor: Use this to paginate through results, utilizing the cursor provided in the previous response.
- order: Sort the results in ascending (ASC) or descending (DESC) order.
- limit: Adjust the page size of results to the app preference.

**Comment by sallymoc** [comment]
This is intended to be used in the explorer, address page (i.e https://explorer.qubic.org/network/address/ZVNDWCSGRJKWKGMRFWFFLIMKKTDDZJANAOQDBEVQMCGXSFVTKBPMIZIDTBZA)

**Comment by alexmf91** [comment]
Would be nice to follow same pagination structure like in epoch ticks endpoint: 
https://qubic.github.io/integration/Partners/qubic-rpc-doc.html#/ArchiveService/ArchiveService_GetEpochTickListV2
```ts
{
  "pagination": {
    "totalRecords": 0,
    "currentPage": 0,
    "totalPages": 0,
    "pageSize": 0,
    "nextPage": 0,
    "previousPage": 0
  },
  "ticks": [ // replacing this with transactions 
    {
      "tickNumber": 0,
      "isEmpty": true
    }
  ]
}
```

**Comment by qubicmio** [comment]
This issue contains several requirements:

* limiting the results by tick range
* limiting the results by date range
* sorting of results by timestamp (desc by default)
* filtering by transaction type
* pagination (2x - cursor based, offset based)

It seems that the endpoint `https://rpc.qubic.org/v2/identities/{identity}/transfers?startTick={startTick}&endTick={endTick}` is used at the moment. Currently 'paging' by tick range seems to be used.

We should discuss (define the requirements for):

* Sorting
* Filtering
* Limiting by date (what for
* Paging

I would suggest to not do that here but discuss this offline first and the create the corresponding issues as we might be able to split them up and implement it iteratively.

If we need a new endpoint is an implementation detail and shouldn't be part of the requirement. Maybe we can modify the current one.

**Comment by sallymoc** [comment]
> It seems that the endpoint https://rpc.qubic.org/v2/identities/{identity}/transfers?startTick={startTick}&endTick={endTick} is used at the moment. Currently 'paging' by tick range seems to be used.
Yes, it's being used in the explorer and web wallet but just because it was the only option to used at the moment to migrate from qubic.li endpoints to qubic rpc. Even when there was a refactoring in the UI to make the UX experience better, solution is not good from my pov:
- It's making too many unneeded requests that in case of having a proper pagination wouldn't be needed. The fact that is not a transaction in a recent tick, doesn't mean that there is no need to ping an older tick, so the app is checking all the ranges possibles to see if there is any transaction.
i.e The explorer page https://explorer.qubic.org/network/address/FAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAYWJB returns at the moment of my testing only 2 results and to load the page it's executes the service more than 60 times with different tick ranges.
- For addresses with many transactions, the request doesn't work. i.e try to run https://rpc.qubic.org/v2/identities/AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFXIB/transfers?startTick=18252273&endTick=18452273 cloudfare returns after some time error 524.
"The Cloudflare 524 error is an indication that Cloudflare successfully established a connection with the origin web server, but did not receive an HTTP response within the default 100-second connection timeout period, as stated by Cloudflare."
- Explorer should display all the actions associated to one identity and not only transfers. Transaction with 0 fees like computors voting or other SC transaction with cost 0, should be also visible when exploring a page. I understand the current service in used is 'transfers' and 'it makes sense' but the problem is that there is not another endpoint to replace this call, that's why this request title is "New endpoint..."
- For the web and mobile wallet apps instead, it makes sense they display only the transactions/events that involve the movement of assets, monetary funds, or ownership rights from one account to another. But still, the existing service is not optimal for this (i.e doesn't return transfers where the identity is one of the 'final' destinations, let's say doesn't take into consideration the internal transfers/events or if at some moment the possession/ownership of assets cost 0, it won't return it)

Let's discuss offline some possible solutions and what would make sense to be consider in application layer vs integration layer.

**Comment by qubicmio** [comment]
This issue will be split into several sub issues.

**Comment by sallymoc** [comment]
@Bertiger @0xluk 
From my last discussion with @qubicmio on this topic, he requested to split this issue (transactions per identity with pagination, filtering results, sorting)
I see this is becoming high relevant now that we have more SC transactions and starting to offering WC integration that we hope is adopted by more DApps. It's not possible to easily see in the explorer the SC transactions for my identities if amount is 0. I would need to search it by trx id or to search in the page, and for that I need to know/store that information.
Can you please consider to implement asap a basic/simple version of this with pagination? filtering, sorting, downloading a file can come later.

**Comment by qubicmio** [comment]
See comment in issue #94 - this is basically true for all sub topics of this issue. It's not possible with the current integration layer infrastructure. We are working on a better query interface. As soon as we have the new infrastructure we can proceed with these feature requests.

**Comment by qubicmio** [comment]
@sallymoc can we close this? RPC 2.0 solves all the issues. See: https://github.com/qubic/archive-query-service/blob/main/v2/README.md

**Comment by sallymoc** [comment]
Closing as per previous comment.
